samples/AzureFunctionsApp/Entities/Lifetime.cs

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.DurableTask.Client;
+using Microsoft.DurableTask.Client.Entities;
+using Microsoft.DurableTask.Entities;
+using Microsoft.Extensions.Logging;
+
+namespace AzureFunctionsApp.Entities;
+
+/// <summary>
+/// Example showing the lifetime of an entity. An entity is initialized on the first operation it receives and then
+/// is considered deleted when <see cref="TaskEntity{TState}.State"/> is <c>null</c> at the end of an operation.
+/// </summary>
+public class Lifetime : TaskEntity<MyState>
+{
+    readonly ILogger logger;
+
+    public Lifetime(ILogger<Lifetime> logger)
+    {
+        this.logger = logger;
+    }
+
+    /// <summary>
+    /// Optional property to override. When 'true', this will allow dispatching of operations to the TState object if
+    /// there is no matching method on the entity. Default is 'false'.
+    /// </summary>
+    protected override bool AllowStateDispatch => base.AllowStateDispatch;
+
+    // NOTE: when using TaskEntity<TState>, you cannot use "RunAsync" as the entity trigger name as this conflicts
+    // with the base class method 'RunAsync'.
+    [Function(nameof(Lifetime))]
+    public Task DispatchAsync([EntityTrigger] TaskEntityDispatcher dispatcher)
+    {
+        this.logger.LogInformation("Dispatching entity");
+        return dispatcher.DispatchAsync(this);
+    }
+
+    public void Init() { } // no op just to init this entity.
+
+    public void CustomDelete()
+    {
+        // Deleting an entity is done by null-ing out the state.
```

Comment:
Are we actually deleting the entity or just clearing its state? I wonder if we can be more explicit about what's happening.